lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -192,6 +192,104 @@ LocaleMatch lookupMatcher(
   return std::u16string(u"uppered");
 }
 
+// Implementation of
+// https://tc39.es/ecma402/#sec-getoption
+// Split into getOptionString and getOptionBool to help readability
+vm::CallResult<Option> getOptionString(
+    const Options &options,
+    const std::u16string &property,
+    llvh::ArrayRef<std::u16string> values,
+    const Option &fallback,
+    vm::Runtime *runtime) {
+  // 1. Assert type(options) is object
+  // 2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  // 3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return Option(fallback);
+  }
+  // 4. Assert: type is "boolean" or "string".
+  // 5. If type is "boolean", then
+  // a. Set value to ! ToBoolean(value).
+  // 6. If type is "string", then
+  // a. Set value to ? ToString(value).
+  // 7. If values is not undefined and values does not contain an element equal
+  // to value, throw a RangeError exception.
+  if (!values.empty() &&
+      llvh::find(values, value->second.getString()) == values.end()) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getString().c_str()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property " +
+            vm::TwineChar16(property.c_str())));
+  }
+  // 8. Return value.
+  return value->second;
+}
+vm::CallResult<Option> getOptionBool(
+    const Options &options,
+    const std::u16string &property,
+    const Option &fallback,
+    vm::Runtime *runtime) {
+  //  1. Assert: Type(options) is Object.
+  //  2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  //  3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return Option(fallback);
+  }
+  //  8. Return value.
```

Comment:
For now, it is okay to do what the Java implementation does, and throw an exception if this is not a bool (by checking using `isBool`)